contrib/packs/actions/workflows/install.yaml

```diff
@@ -1,32 +1,101 @@
----
-  chain:
-    -
-      name: "download pack"
-      ref: "packs.download"
-      parameters:
-        packs: "{{packs}}"
-        force: "{{force}}"
-        python3: "{{python3}}"
-      on-success: "make a prerun"
-    -
-      name: "make a prerun"
-      ref: "packs.virtualenv_prerun"
-      parameters:
-        packs_status: "{{ __results['download pack'].result }}"
-      on-success: "install pack dependencies"
-    -
-      name: "install pack dependencies"
-      ref: "packs.setup_virtualenv"
-      parameters:
-        packs: "{{ __results['make a prerun'].result }}"
-        env: "{{env}}"
-        python3: "{{python3}}"
-      on-success: "register pack"
-    -
-      name: "register pack"
-      ref: "packs.load"
-      parameters:
-        register: "{{register}}"
-        packs: "{{ __results['make a prerun'].result }}"
-
-  default: "download pack"
+version: 1.0
+
+description: A orquesta workflow to install packs.
+
+input:
+  - packs
+  - register
+  - env
+  - force
+  - python3
+  - skip
+
+vars:
+  - packs_list: null
+  - dependency_list: null
+  - conflict_list: null
+  - nested: 3
+
+tasks:
+  init_task:
+    action: core.noop
+    next:
+      - do: download_pack
+
+  download_pack:
+    action: packs.download
+    input:
+      packs: <% ctx().packs %>
+      force: <% ctx().force %>
+      python3: <% ctx().python3 %>
+      dependency_list: <% ctx().dependency_list %>
+    next:
+      - when: <% succeeded() %>
+        do: make_a_prerun
+
+  make_a_prerun:
+    action: packs.virtualenv_prerun
+    input:
+      packs_status: <% task(download_pack).result.result %>
+      packs_list: <% ctx().packs_list %>
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - packs_list: <% task(make_a_prerun).result.result %>
+        do: find_next_action
+
+  find_next_action:
+    action: core.noop
+    next:
+      - when: <% ctx().skip or ctx().nested = 0 %>
+        do: install_pack_requirements
+      - when: <% ctx().nested > 0 and not ctx().skip %>
+        do: get_pack_dependencies
+
+  get_pack_dependencies:
+    action: packs.get_pack_dependencies
+    input:
+      packs_status: <% task(download_pack).result.result %>
+      nested: <% ctx().nested%>
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - dependency_list: <% task(get_pack_dependencies).result.result.dependency_list %>
+          - conflict_list: <% task(get_pack_dependencies).result.result.conflict_list %>
+          - nested: <% task(get_pack_dependencies).result.result.nested %>
+        do: check_dependency_and_conflict_list
+
+  check_dependency_and_conflict_list:
+    action: core.noop
+    next:
+      - when: <% ctx().conflict_list %>
+        do: stop_installation_and_cleanup
+      - when:  <% not ctx().conflict_list and ctx().dependency_list %>
+        do: download_pack
+      - when: <% not ctx().conflict_list and not ctx().dependency_list %>
+        do: install_pack_requirements
```

Comment:
Say a dependency is already downloaded and installed on the system separately before this workflow is run.  What if there is a conflict?  Will the packs.delete also delete this dependency that was installed separately?

Reply:
@m4dcoder No, it is not deleted. Workflow works as checking for dependency and conflict packs before download dependency packs. If  conflict occurs,  workflow will delete only packs that is already installed for `this` workflow not separately before this workflow.

Reply:
Please make sure there is a test case for this.

Reply:
> Please make sure there is a test case for this.

Sure, this test case should be a `e2e` test.

Reply:
Btw, have you actually tested this manually and see that a previously installed pack is unaffected by the clean up step here?

Reply:
> Btw, have you actually tested this manually and see that a previously installed pack is unaffected by the clean up step here?

Yes, I did test it manually.  Actually I manually tested lots of corner cases.

Reply:
These corner cases should be automated in the list of test cases below.

Reply:
Test added in https://github.com/StackStorm/st2tests/pull/179